pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -243,10 +246,17 @@ private String getQueryResponse(String query, @Nullable SqlNode sqlNode, String
     // Get resource table name.
     String tableName;
     try {
-      String inputTableName =
-          sqlNode != null ? RequestUtils.getTableNames(CalciteSqlParser.compileSqlNodeToPinotQuery(sqlNode)).iterator()
-              .next() : CalciteSqlCompiler.compileToBrokerRequest(query).getQuerySource().getTableName();
-      tableName = _pinotHelixResourceManager.getActualTableName(inputTableName);
+      if (sqlNode != null) {
+        PinotQuery pinotQuery = CalciteSqlParser.compileSqlNodeToPinotQuery(sqlNode);
+        if (pinotQuery.isShowTables()) {
+          return showTablesResponse();
+        }
+        String inputTableName = RequestUtils.getTableNames(pinotQuery).iterator().next();
+        tableName = _pinotHelixResourceManager.getActualTableName(inputTableName);
```

Comment:
this is for pinot controller only, what if the query is sent to pinot broker?
I feel you can call getTables API for response then wrap it as query response.

Reply:
@xiangfu0 I've moved code to broker now. Can you have a look?